backend/src/routes/club.js

```diff
@@ -18,124 +16,149 @@ router.get("/", async (req, res) => {
   }
 });
 
-// Returns a club with the given id
-
 router.get("/:id", async (req, res) => {
   console.log("GET /club/:id");
-  let data = {};
+  let clubData = {};
   try {
     const conn = await pool.getConnection();
-    const club = await conn.query("SELECT * FROM Club WHERE id = ?", [
+
+    const club = await conn.query("SELECT * FROM club WHERE id = ?", [
       req.params.id,
     ]);
-    data.club = club[0];
-    const teams = await conn.query("SELECT * FROM Team WHERE clubId = ?", [
+    if (club.length === 0) {
+      return res.status(404).json({ message: "Club not found" });
+    }
+    clubData.club = club[0];
+
+    const teams = await conn.query("SELECT * FROM team WHERE clubId = ?", [
       req.params.id,
     ]);
-    data.teams = teams;
-    console.log(data);
-    res.json(data);
+    clubData.teams = teams;
+
+    console.log(clubData);
+    res.json(clubData);
   } catch (err) {
     console.log(err);
-    res.status(500).json({ message: err.message, errorCode: err.errno });
+
+    res.status(500).json({
```

Comment:
Funderar på att sätta in emuns, så det blir samma meddelande för varje 500 osv. Bara en tanke.

Reply:
Det är bättre med enums men tror inte vi hinner med det.